Formula/synergy-core.rb

```diff
@@ -0,0 +1,376 @@
+           "permission to the terminal emulator."
+      if Dir.exist?("/System/Applications/Utilities/Terminal.app")
+        s += " For example, if you use Terminal.app, you must\ngrant the 'Accessibility' permission to:\n" \
+             "  /System/Applications/Utilities/Terminal.app"
+      end
+      s += "\n\n"
+      if Hardware::CPU.arm?
+        s += "On Apple Silicon, if you run Synergy under a shell (such as bash), you may also\n" \
+             "need to grant the 'Accessibility' permission to that shell.\n\n"
+      end
+      s += "Regardless of how you run Synergy, the following applies:\n" \
+           "  Each time you upgrade Synergy, you must manually remove the relevant applications\n" \
+           "  from the 'Accessibility' permissions page and then re-add them. If you skip this step,\n" \
+           "  Synergy may not work after being upgraded.\n\n" \
+           "  It is not adequate to untick the checkbox beside each relevant application. You must\n" \
+           "  remove each application from the list using the '-' button before re-adding them.\n\n" \
+           "  Alternatively, run this command to clear the 'Accessibility' permissions page:\n" \
+           "    sudo tccutil reset Accessibility\n" \
+           "  After running this command, you will need to re-add programs that you want in the list.\n\n" \
+           "  Make sure that you exit Synergy before revoking the 'Accessibility' permission. If\n" \
+           "  Synergy is running when the permission is revoked, you may experience highly anomalous\n" \
+           "  behavior with your mouse and keyboard."
+      if Dir.exist?("/Applications/Synergy.app")
+        s += "\n\n" \
+             "If you previously installed Synergy without using this formula, you may want to\n" \
+             "delete your previous installation to avoid confusion. It appears that you have a\n" \
+             "previous non-brew installation of Synergy at /Applications/Synergy.app. You may\n" \
+             "experience problems granting the 'Accessibility' permission to both copies of\n" \
+             "Synergy at once, even if they aren't running at the same time."
+      end
+      return s
+    end
+  end
+
+  test do
+    port = free_port
+    conf_file = testpath/"synergy.conf"
+    conf_file.write <<~EOS
+      section: screens
+        test-server:
+        test-client:
+      end
+    EOS
+
+    # The Synergy client and server print messages with two formats:
+    #   [2022-04-24T15:18:27] NOTE: started client
+    # and
+    #    2022-04-24 15:18:27.816 synergyc[57673:302843] starting cocoa loop
+    #
+    # In both cases, the message text starts after the first ']' character.
+    log_trim = ->(log_line) { log_line.split("]", 2)[1]&.strip! }
+
+    def kill_and_wait_for(process)
+      Process.kill("KILL", process.pid)
+      Process.wait(process.pid)
+    end
```

Comment:
This should just be repeated where needed. Functions make tests very hard to understand/read

Reply:
Done.